resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -110,37 +108,41 @@ private[spark] class Client(
         .done()
       try {
         val selectors = Map(DRIVER_LAUNCHER_SELECTOR_LABEL -> driverLauncherSelectorValue).asJava
-        val (servicePorts, containerPorts) = configurePorts()
+        val containerPorts = configureContainerPorts()
+        val driverLauncherServicePort = new ServicePortBuilder()
```

Comment:
I think we should delay the creation of the service as much as possible; meaning that we should wait till after the pod is submitted and running. Also, we should make a decision as to whether there are local jars being submitted or not. If there aren't, we don't need to expose the NodePort at all. WDYT?